hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/metrics/ContractMetrics.java

```diff
@@ -74,77 +126,164 @@ public class ContractMetrics {
+                final var functionalityName = POSSIBLE_FAILING_TX_TYPES.get(txKind) + "DueToIntrinsicGas";
+                final var name = toRejectedName(functionalityName, REJECTED_FOR_GAS_SHORT_DESCR);
+                final var descr = toRejectedDescr(functionalityName, REJECTED_FOR_GAS_SHORT_DESCR);
+                final var config = new Counter.Config(METRIC_CATEGORY, name)
+                        .withDescription(descr)
+                        .withUnit(METRIC_TXN_UNIT);
+                final var metric = newCounter(metrics, config);
+                rejectedTxsLackingIntrinsicGas.put(txKind, metric);
+            }
 
-                final var contractsConfig = requireNonNull(contractsConfigSupplier.get());
-                this.p1MetricsEnabled = contractsConfig.metricsSmartContractPrimaryEnabled();
+            // Rejected transactions for ethereum calls that are in type 3 blob transaction format
+            {
+                final var name = toRejectedName(REJECTED_TYPE3_FUNCTIONALITY, REJECTED_TYPE3_SHORT_DESCR);
+                final var descr = toRejectedDescr(REJECTED_TYPE3_FUNCTIONALITY, REJECTED_TYPE3_SHORT_DESCR);
+                final var config = new Counter.Config(METRIC_CATEGORY, name)
+                        .withDescription(descr)
+                        .withUnit(METRIC_TXN_UNIT);
+                final var metric = newCounter(metrics, config);
+                rejectedEthType3Counter = metric;
+            }
+        }
+    }
 
-                final var metrics = requireNonNull(metricsSupplier.get());
+    private @NonNull Counter makeCounter(
+            @NonNull final MethodMetricType metricType,
+            @NonNull final String nam,
+            @NonNull final String clarification) {
+        final var name = toMethodMetricName(nam, metricType);
+        final var descr = toMethodMetricDescr(nam, metricType, clarification);
+        final var config =
+                new Counter.Config(METRIC_CATEGORY, name).withDescription(descr).withUnit(METRIC_TXN_UNIT);
+        return newCounter(metricsSupplier.get(), config);
+    }
 
-                if (p1MetricsEnabled) {
-                    // Rejected transactions counters
-                    for (final var txKind : POSSIBLE_FAILING_TX_TYPES.keySet()) {
-                        final var name = toRejectedName(txKind, REJECTED_TXN_SHORT_DESCR);
-                        final var descr = toRejectedDescr(txKind, REJECTED_TXN_SHORT_DESCR);
-                        final var config = new Counter.Config(METRIC_CATEGORY, name)
-                                .withDescription(descr)
-                                .withUnit(METRIC_TXN_UNIT);
-                        final var metric = newCounter(metrics, config);
-                        rejectedTxsCounters.put(txKind, metric);
-                    }
+    private @NonNull Counter[] makeCounterPair(@NonNull final String nam, @NonNull final String clarification) {
```

Comment:
there seems to be many instances where I think nam -> name.  Unless there is a reason?